addons/website/static/tests/tours/snippet_image_gallery.js

```diff
@@ -47,11 +47,8 @@ tour.register("snippet_image_gallery", {
     content: "Click on move to next",
     trigger: ".snippet-option-gallery_img we-button[data-position='next']",
 }, {
-    content: "Click on the moved image",
-    trigger: "#wrap .s_image_gallery .carousel-item.active img[data-index='1']",
-}, {
-    content: "Check that the image still has the correct filter",
-    trigger: ".snippet-option-ImageOptimize we-select:contains('Filter') we-toggler:contains('Blur')",
+    content: "Check that the moved image still has the correct filter",
+    trigger: ".s_image_gallery .carousel-item.active img[data-index='1'][data-gl-filter='blur']",
```

Comment:
Specs:

> The tour should probably both check the image still has the filter AND the filter option properly show the right value.

=> Let's do that please. Checking the UI is actually the point of my spec: in 16.3 (maybe here too?) if you revert your previous fix, the UI does **not** display that Filter option but the step **succeeds** (like you said in your commit message). I don't understand why and I want:
- A check of the UI to be made after the move
- That check fails without your previous fix

\+ I don't understand your commit explanation: why is the current version in 15.0 a cause of undeterministic error? You move the image, then you click on it using the `[data-index='1']` selector, meaning the move is finished, then you check the right panel. Is not this right?

Reply:
> I don't understand your commit explanation: why is the current version in 15.0 a cause of undeterministic error? 

@qsm-odoo , here is how I understand this situation (but do not hesitate to correct me if I am wrong :) ); The problem in this test is that we are checking a certain condition (that the blur filter is displayed in the option panel) that is true even before a certain action (the move of the image) has been done. I think that this can lead to undeterministic results. Indeed, in this case, if the condition is true after the action has been done, we are not sure that it represents a stable state where the consequences of the action (in this case, the reload of the UI) have been completely applied; it could also be the old state that still has to evolve due to the action. I think that it is what is explicitly happening in the version 16.3; the second check of the UI is true because the snippet editor has probably not have enough time to reload. However, after a certain delay, the UI changes and the condition is false. Note that this could also theoretically happen in this version but, because the delays are not the same in version 15 and 16.3, it is not explicitly visible in version 15.
To avoid this uncertainty, I modified the tour to test a condition that is `false` before the action (there is no filter on `img[data-index='1']` before the move of the image) and is `true` after the action.
It also explains why I removed the check on the UI (as it is not a sufficient condition). But more than that, I think that if I add it back (even after the check "Check that the moved image still has the correct filter"), we will in fact never be sure if the condition checks an old unstable state of the UI or the new (stable) one.

This also explains why it is not sufficient to click on the moved image before checking the UI: The move action is finished but we are not sure that the UI is in a stable state as the condition that we check is the same before and after the action.

Reply:
> that is true even before a certain action (the move of the image) has been done.

Yes, maybe. Did you test adding a step (clicking elsewhere) in the page and then clicking on the image? In that case, there is no reason for the test to succeed... still it passed when I tested. Whatever the reason is: the test should be fixed to test the UI.

> Note that this could also theoretically happen in this version but, because the delays are not the same in version 15 and 16.3, it is not explicitly visible in version 15.

Then something unexpected changed: this is a bug and it has to be fixed and tested.

> To avoid this uncertainty, I modified the tour to test a condition that is false before the action

That's what I did in 16.4 to fix the issue indeed. But I added a FIXME, since there is indeed a change that occurred between 15.0 and 16.3, that apparently made the UI less consistent (even if it is an nondeterministic issue like you said, it apparently became more nondeterministic): I want to test the UI too.

> we will in fact never be sure if the condition checks an old unstable state of the UI or the new (stable) one

We can never be sure of anything, true. We can still try to test the closest thing we can. We normally do not show the UI unless it is updated, so if you click elsewhere, go back to the image and check the UI... it should test what you want. And having that in the tour helped: it showed that in 16.3 something happened which broke that (even if nondeterministic, but when I tested, it really seemed systematic).